src/quo2/components/avatars/user_avatar.cljs

```diff
@@ -84,71 +76,70 @@
    :height        inner-dimensions
    :border-radius inner-dimensions})
 
+(defn outer-styles
+  [outer-dimensions]
+  {:width         outer-dimensions
+   :height        outer-dimensions
+   :border-radius outer-dimensions})
+
 (defn container
-  [inner-dimensions outer-dimensions & children]
-  [rn/view
-   {:style (merge {:background-color (colors/custom-color-by-theme :turquoise 50 60)
-                   :justify-content  :center
-                   :align-items      :center}
-                  (container-styling inner-dimensions outer-dimensions))}
-   children])
+  [{:keys [inner-dimensions outer-dimensions]} & children]
+  (-> [rn/view
+       {:style (merge
+                {:background-color (colors/custom-color-by-theme :turquoise 50 60)
+                 :justify-content  :center
+                 :align-items      :center}
+                (container-styling inner-dimensions outer-dimensions))}]
+      (concat children)
+      (vec)))
+
+(def one-inital-letter-sizes #{:xs :xxs :xxxs})
+(def valid-ring-sizes #{:big :medium :small})
 
-(def small-sizes #{:xs :xxs :xxxs})
-(def identicon-sizes #{:big :medium :small})
+(defn initials-avatar
+  [{:keys [full-name size inner-dimensions outer-dimensions]}]
+  (let [amount-initials (if (one-inital-letter-sizes size) 1 2)
+        initials        (->> (split full-name " ")
+                             (transduce (comp (map first) (take amount-initials))
+                                        str)
+                             (upper-case))
+        font-size       (get-in sizes [size :font-size])]
```

Comment:
It's probably best to avoid transducers because they're not as commonly understood as map/filter/reduce:
``` clojure
(->> (split full-name ")
     (map (comp first upper-case))
     (take amount-initials)
     (clojure.string/join))
```

Reply:
It's also recommended to use `string/upper-case` and `string/join` here instead of `upper-case` and `join`.

Reply:
I agree with @erikseppanen, but for a different reason.

I think the original implementation is mixing transducers with the normal pipelining macro, but IME this often complicates things. It's similar to code that mixes thread-first and thread-last, sometimes nested. I think it's more idiomatic to pick one style for each kind of data processing.

On the other hand, I'm actually on the camp that think transducers should be used far more often, especially for processing larger collections, since they perform better and produce less intermediate data, thus putting less pressure on the GC. For small collections there's no meaningful difference. I can imagine certain screens and subscriptions could take advantage of this. There are good benchmarks out there proving there can be substantial differences, so maybe, just maybe, in a resource constrained environment such as mobile, the performance improvements could be significant.

Reply:
I agree @ilmotta, for processing large collections. I had a problem exactly like that once, and looked at both transducers and reducers, and just used reducers for that. Also agree with your other points!

Reply:
@erikseppanen @ilmotta 
Well, I used a transducer because it's familiar to me, but  I agree with @ilmotta about mixing thread-last and transduce sometimes makes code more complicated.

In this case it doesn't wort to use transducers, it's removed in af3d6dec6ce6082a6555029ab069f20dcfb7c8fb, I used an `as->>` since I was not confortable with that  `(split full-name " ")`  as parameter of `->>`.

For large collections, it'd be nice to use transducers :)

Reply:
Just a quick addendum here, but the whole `initials` binding could become a function instead. This would help with RDD too. That implementation is a bit too dense to live in a `let` binding.